meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh

```diff
@@ -11,14 +11,12 @@ DST_LINK_FILE_BASE="/run/systemd/network/"
 mkdir -p "${DST_ENVS_BASE}"
 mkdir -p "${DST_LINK_FILE_BASE}"
 
-# Set the static hostname if it is not set yet
-# (The transient hostname is passed via systemd.hostname= commandline)
-# --------------------------------------------------------------------
+# If no hostname is set yet persist the one we currently have
+# -----------------------------------------------------------
 
 if [[ ! -e /etc/hostname ]]
 then
-    hostname="$(hostnamectl --transient hostname)"
-    hostnamectl --static hostname "${hostname}"
+    hostname > /etc/hostname
```

Comment:
This change deserves to be mentioned in the commit message.

Reply:
Split into a separate commit with some info in the commit message.
Also added more information in the comment.